### ISSUE TITLE
Unresponsive page

### ISSUE THREAD

**Issue by YashdalfTheGray** [synergistic-silobusters/flintandsteel#309]
Um...I don't even know what to say here...maybe this is only on dev but,

Steps to reproduce -
1. Open up flintandsteel in a tab
2. Interact with it, do something on the page, login, click around, whatever.
3. Open up another flintandsteel tab
4. Click around on that new tab

Result: the new tab is totally unresponsive...until, the old tab is closed and then it responds in a flash and goes back to being frozen again. 

The only fix is to not open 2 tabs on the website or to close all of the tabs down and reopen just one. 

It's bizarre, I'm hoping it's a one off but all the code that I have is doing it.

**Comment by Mctalian** [comment]
I also noticed this tonight. When you look at the "network" tab in a chrome dev window, does it show a few files as "pending"?

**Comment by YashdalfTheGray** [comment]
I didn't check the networks tab but I could make a guess that it's waiting on something. That seems to fit the description of something that it would be waiting on.

**Comment by Mctalian** [comment]
Can you check and see if you're also experiencing that?

**Comment by davethenipper** [comment]
I've used two tabs on production before (about 2-3 weeks ago) and it was
working fine.  Not sure if this changed.

**Comment by Mctalian** [comment]
Yeah, I think production was fine as well. We'll have to back track and pin
down which commit.

Happy "Go-Live" Week! :P

**Comment by alanlgirard** [comment]
Just confirmed that this is currently not an issue for production as it is currently so thats good, but we need to track down the issue before we push the next update to prod

**Comment by Mctalian** [comment]
This *might* only be an issue on Chrome? I forgot about this and pushed all the recent commits to prod, and now I'm trying to track this down.

**Comment by Mctalian** [comment]
I've tracked it back to commit 4445803 being the last working version. The next commit is replacing gif files so I'm not really sure how that would cause anything like that.

**Comment by Mctalian** [comment]
But at commit 4445803, opening 3 tabs causes the issue.

Chrome only allows 6 (I think) concurrent connections to the same server at a given time... so I think this has something to do with the number of files we're trying to send at once in conjunction with each tab having a few SSE connections as well.

So I think we always had this issue, but it's getting worse overtime. This is something we'll have to look at closely to find a solution.

**Comment by Mctalian** [comment]
Ok, I figured out why this came out of nowhere.

A few devs have been adding "SidenavCtrl" to random places in the html, so it's instantiating the controller multiple times. Therefore, just navigating to the page, we are creating *3* SSEs for the sidenav. It used to be that at any given time, we'd only have 2 SSEs possible: one for the sidenav in case an idea is edited or liked, and one for the idea you are currently viewing OR if you're on the Browse page, it would be a similar SSE to the navbar.

Long story short, there are 2 optimizations I can make to get this back to the good ol' days which will buy us time before socket.io. Also, the Homepage redesign will knock out the Sidenav SSE so we would only have 1 active connection at any given time for SSEs which might be "good enough" for a while.

Either way, should have something soon.

**Comment by YashdalfTheGray** [comment]
You can run a blame on the file to see who's been doing it. This might repeat, they need to be told. We can do it privately, but this should be pointed out. I'm assuming you already know who it is, just send them a note.
